src/Cassandra.IntegrationTests/TestBase/TestGlobals.cs

```diff
@@ -184,6 +185,29 @@ public static bool VersionMatch(TestCassandraVersion versionAttr, Version execut
             return comparison == versionAttr.Comparison;
         }
 
+        public static async Task Connect(Cluster cluster, bool asyncConnect, Action<ISession> action)
+        {
+            if (asyncConnect)
+            {
+                try
+                {
+                    var session = await cluster.ConnectAsync();
+                    action(session);
+                }
+                finally
+                {
+                    cluster?.ShutdownAsync().Wait();
```

Comment:
I don't think its a good idea to reuse this method for the memory tests.
If awaiting for `ConnectAsync()`, we should `await` for `ShutdownAsync()` (instead of wait).

Reply:
I am removing this dotMemoryUnit test as it is not possible to run this test async (dotMemory loose profiling).